src/components/ActivityFeed/ActivityFeedItem.js

```diff
@@ -186,29 +198,75 @@ export class ActivityFeedItem extends React.Component<Props> {
     return (smartWalletStatus.status !== SMART_WALLET_UPGRADE_STATUSES.DEPLOYMENT_COMPLETE);
   }
 
+  findAccount = (address: string) => {
+    const { accounts } = this.props;
+    return accounts.find(account => addressesEqual(account.id, address));
+  }
+
+  isSWAddress = (address: string) => {
+    const account = this.findAccount(address);
+    return (account && account.type === ACCOUNT_TYPES.SMART_WALLET);
+  }
+
+  isKWAddress = (address: string) => {
+    const account = this.findAccount(address);
+    return (account && account.type === ACCOUNT_TYPES.KEY_BASED);
+  }
+
+  getFormattedSettleValues = () => {
+    const {
+      event,
+      asset,
+      assets,
+      supportedAssets,
+    } = this.props;
+    const settleData = event.extra;
+    const ppnTransactions = asset
+      ? settleData.filter(({ symbol }) => symbol === asset)
+      : settleData;
+
+    const valueByAsset: Object = {};
+
+    ppnTransactions.forEach((trx) => {
+      const { symbol, value: rawValue } = trx;
+      const { decimals = 18 } = getAssetData(assets, supportedAssets, symbol);
+      const value = new BigNumber(rawValue);
+      if (!valueByAsset[symbol]) {
+        valueByAsset[symbol] = { ...trx, value, decimals };
+      } else {
+        const { value: currentValue } = valueByAsset[symbol];
+        valueByAsset[symbol].value = currentValue.plus(value);
+      }
+    });
+
+    const valuesArray = Object.keys(valueByAsset).map((key) => valueByAsset[key]);
```

Comment:
What do you think about that approach?
```suggestion
    const valuesArray = (Object.values(valueByAsset): any);
```